.github/workflows/publish.yml

```
@@ -0,0 +1,30 @@
+name: Publish Crate
+
+on:
+  push:
+    tags:
+      - '*'  # Trigger when a tag is created
```

minor: I usually prefer triggering release when a GitHub release is created:
```suggestion
on:
  release:
    types: [published]
```